src/generators/default_map_generator_job.cpp

```diff
@@ -785,17 +762,26 @@ std::string default_map_generator_job::default_generate_map(size_t width, size_t
 	// If the [naming] child is empty, we cannot provide good names.
 	std::map<map_location,std::string>* misc_labels = naming.empty() ? nullptr : labels;
 
-	boost::shared_ptr<name_generator> name_generator;
-	if(naming.has_attribute("name_generator")) {
-		name_generator.reset(new context_free_grammar_generator(naming["name_generator"]));
-		if(!name_generator->is_valid()) {
-			name_generator.reset();
-		}
-	}
-	config::attribute_value markov_list = naming.get_old_attribute("names", "male_names",
-		"[naming]male_names is deprecated, use names instead");
-	if(!markov_list.blank()) {
-		name_generator.reset(new markov_generator(utils::split(markov_list), naming["markov_chain_size"], 12));
+	name_generator_factory* generator_factory = nullptr;
+	name_generator	*base_name_generator      = nullptr,
+					*river_name_generator     = nullptr,
```

Comment:
Personally, I would prefer if you repeat the type name on each line here, rather than declaring several pointers in a single statement. I'm not sure if Wesnoth has an official stance on this though.